lib/jira/commands/log.rb

```diff
@@ -3,11 +3,107 @@ class CLI < Thor
 
     desc "log", "Logs work against the input ticket"
     def log(ticket=Jira::Core.ticket)
-      time_spent = self.io.ask("Time spent on #{ticket}")
+      time_spent = self.io.ask("Time spent on ticket #{ticket}")
       self.api.post("issue/#{ticket}/worklog", { timeSpent: time_spent }) do |json|
-        puts "Successfully logged #{time_spent} on #{ticket}."
+        puts "Successfully logged #{time_spent} on ticket #{ticket}."
       end
     end
 
+    desc "logd", "Deletes work against the input ticket"
+    def logd(ticket=Jira::Core.ticket)
+      if self.io.agree("List worklogs for ticket #{ticket}")
+        logs(ticket)
+      end
+      idx = self.get_worklog_idx("delete")
+      if idx < 0
+        puts "No worklog deleted."
+        return
+      end
+      self.api.get("issue/#{ticket}/worklog") do |json|
+        worklogs = json['worklogs']
+        if idx < worklogs.count
+          id = worklogs[idx]['id']
+          time_spent = worklogs[idx]['timeSpent']
+          self.api.delete("issue/#{ticket}/worklog/#{id}") do |json|
+            puts "Successfully deleted #{time_spent} on ticket #{ticket}"
+            return
+          end
+        end
+      end
+      puts "No worklog deleted."
+    end
+
+    desc "logs", "Lists work against the input ticket"
+    def logs(ticket=Jira::Core.ticket)
+      self.api.get("issue/#{ticket}/worklog") do |json|
+        worklogs = json['worklogs']
+        if worklogs.count > 0
+          worklogs.each do |worklog|
+            author = worklog['author']['displayName']
+            time = Time.parse(worklog['updated'])
+            time_spent = worklog['timeSpent']
+
+            printf "[%2d]", worklogs.index(worklog)
+            puts "  #{Jira::Format.user(author)} @ "\
+                 "#{Jira::Format.time(time)}:\n"\
+                 "#{Jira::Format.comment(time_spent)}"
+          end
+        else
+          puts "There are no worklogs on ticket #{ticket}"
+        end
+      end
+    end
+
+    desc "logu", "Updates work against the input ticket"
+    def logu(ticket=Jira::Core.ticket)
+      if self.io.agree("List worklogs for ticket #{ticket}")
```

Comment:
Looking at this logic, it can benefit a lot from breaking the logic out into helper methods.
I've been planning on restructuring the core infrastructure of this gem so each of the commands can be written as a class rather than having all the logic of each within the same namespace. No action needed for now, but this is a heads up.